docs/conceptual/iommu.rst

```diff
@@ -0,0 +1,98 @@
+.. meta::
+   :description: Input-Output Memory Management Unit (IOMMU)
+   :keywords: IOMMU, DMA, PCIe, XGMI, AMD, ROCm
+
+****************************************************************
+Input-Output Memory Management Unit (IOMMU)
+****************************************************************
+
+Overview
+================================================================
+
+The IOMMU (IO Memory Management Unit) is a device which provides memory remapping services to 
+IO devices on a platform. Similar to the MMU on a CPU, the IOMMU is able to provide a private 
+virtual address space for each device or group of devices on a platform. While the IOMMU is 
+most commonly used for virtualization provide a private IO space for devices used in a guest 
+virtual machine, it can also be used to isolate devices on bare metal to prevent accidental
+or nefarious DMA accesses to things that they should should not have access to. The IOMMU provides
+virtualized access to the entire platform address space, not just system memory. It can also
+provide virtualized access to the MMIO portions of the platform address space. This is used for
+things like peer to peer DMA. In addition to DMA remapping, the IOMMU also support interrupt remapping.
+In addition to virtualization use cases, this can be advantageous for devices which support multiple
+interrupts on bare metal and it may be required for interrupt delivery on platforms with lots of cores.
+
```

Comment:
I only see Peer to Peer DMA in the topics. Is that what we want to emphasize?

Reply:
The IOMMU is required to support multiple interrupts per device on devices which support MSIs in Linux.  Supporting multiple interrupts per device without the IOMMU requires MSI-X which not all devices support.

Reply:
But is that something we want to emphasize? Do we want a separate topic for interrupt remapping?

Reply:
It provides context as to why you may want some level of IOMMU enabled.  E.g., we may have users that have a system with a lot of cores and some NICs that need multiple MSIs to perform optimally.  These requirements would require IOMMU, but at the same time, they don't want the extra overhead of the IOMMU for P2P DMA between GPUs.  Setting iommu=pt would provide a middle ground by allowing the interrupt remapping, but not using device IO remapping.